commands/push-image.ts

```diff
@@ -1,14 +1,29 @@
 import vscode = require('vscode');
 import { ImageItem, quickPickImage } from './utils/quick-pick-image';
 import { reporter } from '../telemetry/telemetry';
+import { DockerNode } from '../explorer/dockerExplorer';
 const teleCmdId: string = 'vscode-docker.image.push';
 
-export async function pushImage() {
+export async function pushImage(context?: DockerNode) {
+    let imageToPush: Docker.ImageDesc;
```

Comment:
Yes, on line 22.  if the user never selects anything from the quick pick then its undefined.  The alternative is to test `imageName`, but the `imageToPush` pattern is more consistent across the other commands.